src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.updateAliveList();
+    }
+  }
+
+  static updateAliveList() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+      target.checkHealth();
```

Comment:
[CODE PERFORMANCE] - While the `bite` method correctly filters out non-herbivore targets and hidden herbivores, the subsequent call to `target.checkHealth` within the `if` block could lead to performance issues if there are many animals in the `Animal.alive` array. Consider moving the `Animal.updateAliveList` call outside of the `checkHealth` method and invoke it only when necessary, for example, after the `bite` method has been executed on all targets.